File: Library-Client-App/src/main/resources/static/js/trainee/search-project.js
let searchProject = new Object();
let table = null;

$(document).ready(() => {
    getAll();
});

function getAll() {
    table = $('#cariTabel').DataTable({
        filter: true,
        orderMulti: true,
        ajax: {
            url: "/project/cari",
            datatype: "json",
            dataSrc: ""
        },
        columns: [
            {
                data: "idProject", name: "ID Project", autoWidth: true
            },
            {
                data: "judul", name: "Judul", autoWidth: true
            },
            {
                data: "batch", name: "Batch", autoWidth: true
            },
            {
                data: "nama", name: "Nama Trainee", autoWidth: true
            },
            {
                data: "trainer", name: "Nama Trainer", autoWidth: true
            },
            {
                data: "deskripsi", name: "Deskripsi", autoWidth: true
            },
            {
                data: "skema", name: "Deskripsi", autoWidth: true
            },
            {
                data: "link", name: "Link Full Project", autoWidth: true
            },
            
        ]
    });
}

File: Library-Client-App/src/main/java/co/id/library/project/LibraryClientApp/models/SearchProject.java

package co.id.library.project.LibraryClientApp.models;

import java.util.List;
import lombok.Data;

@Data
public class SearchProject {
    private Integer idProject;
    private String judul;
    private String batch;
    private List<String> nama;
    private String trainer;
    private String deskripsi;
    private String skema;
    private String link;

    public SearchProject(Integer idProject, String judul, String batch, List<String> nama, 
            String trainer, String deskripsi, String skema, String link) {
        this.idProject = idProject;
        this.judul = judul;
        this.batch = batch;
        this.nama = nama;
        this.trainer = trainer;
        this.deskripsi = deskripsi;
        this.skema = skema;
        this.link = link;
    }

    public SearchProject() {
    }
   
  
    
    
}